mypy/plugins/dataclasses.py

```diff
@@ -188,6 +190,9 @@ def transform(self) -> None:
         else:
             self._propertize_callables(attributes)
 
+        if decorator_arguments['slots']:
+            self.add_slots(info, attributes, py_version >= (3, 10))
```

Comment:
```suggestion
            self.add_slots(info, attributes, correct_version=py_version >= (3, 10))
```
I'd also mark the param keyword-only in the function definition.